### PR TITLE
Fix issue with mising dashboards for new scylla versions

### PR DESCRIPTION
**Pull request by aleksbykov**
For branch-3.2 there is no appropriate dashboards in scylla-monitoring
repo. So sct installs the dashboards for master. This cause and error
that dashboard doesn't exists and screenshots and snapshots doesn't
collected correctly.

Fix added the check, that appropriate dashboard exists in Grafana,
and if not, use 'master' instead of scylla version for dashboard.

Also, code for geting correct screenshot resolution was changed

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/logcollector.py [hunk 1]
```diff
@@ -281,6 +281,31 @@ def get_grafana_annotations(self, node):  # pylint: disable=inconsistent-return-
             LOGGER.warning("unable to get grafana annotations [%s]", str(ex))
             return ""
 
+    @staticmethod
+    def dashboard_exists(node: CollectingNode, uid: str) -> bool:
```
Comment (fruch) [inline sdcm/logcollector.py]: kudos for using hinting !!!

File: sdcm/logcollector.py [hunk 2]
```diff
@@ -281,6 +281,31 @@ def get_grafana_annotations(self, node):  # pylint: disable=inconsistent-return-
...
+        """Check on Grafana server, that dashboard exists
+
+        Send request to Grafana and validate that dashboard with uid
+        provided as "uid" parameter is available and could be requested
+
+        Arguments:
```
Comment (fruch) [inline sdcm/logcollector.py]: as for docs, my personal preference is the reST/sphinx format and not the google one
https://stackabuse.com/python-docstrings/

but having some would always help

File: sdcm/logcollector.py (hunk 1 above)
Comment (bentsi) [reply sdcm/logcollector.py]: Like for type annotations!

File: sdcm/logcollector.py (hunk 2 above)
Comment (bentsi) [reply sdcm/logcollector.py]: we haven't defined the "official" docstring format, let's discuss it on our QA meeting. Let's leave as is.

File: sdcm/logcollector.py (hunk 1 above)
Comment (bentsi) [reply sdcm/logcollector.py]: but, we can't backport to 3.2/3.1 since the branches are not running using Python 3

**Comment by bentsi** [comment]
@aleksbykov we need this on `branch-3.2`. Can you remove type hints so it will work on Python 2?

**Comment by aleksbykov** [comment]
@bentsi , i think branch-3.2 running with python3.

**Comment by aleksbykov** [comment]
@fruch What does LGTM mean ? )))

**Comment by bentsi** [comment]
@aleksbykov 
- python 3 support is in master only 
- LGTM - looks good to me

File: sdcm/logcollector.py (hunk 1 above)
Comment (fruch) [reply sdcm/logcollector.py]: we could change it on backport, but we need to pay close attention while doing so.
i.e. I think for anything that needs backport, lets not use python3 only features just yet.

Comment (aleksbykov) [reply sdcm/logcollector.py]: fixed for backport

File: sdcm/logcollector.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/logcollector.py]: leave as is

**Comment by aleksbykov** [comment]
@fruch remove python3 specific features

**Comment by fruch** [comment]
@aleksbykov branch-3.1 doesn't have log collector

**Pull request merged**